betty/requirement.py

```diff
@@ -238,10 +245,58 @@ def requires_project(
     async def _requires_project(
         cls: type[_HasRequirementT], service_level: ServiceLevel
     ) -> Requirement | None:
-        from betty.project import Project
-
         if not isinstance(service_level, Project):
             return StaticRequirement(Plain("A Project is required."))
         return await f(cls, service_level)
 
     return _requires_project
+
+
+def requires_extension(
```

Comment:
Can we extract common functionality? E.g. as a base class for the three decorators?